packages/core/src/html/SiteLinkManager.ts

```diff
@@ -69,4 +74,48 @@ export class SiteLinkManager {
     this._addToCollection(resourcePath, cwf);
     return 'Intralink collected to be validated later';
   }
+
+  maintainFilePathToHashesMap(node: MbNode, cwf: string, forceWrite: string = '') {
+    if (!this.config.intrasiteLinkValidation.enabled) {
+      return;
+    }
+    const path = cwf.substring(this.config.rootPath.length);
+    if (!this.filePathToHashesMap.has(path)) {
+      this.filePathToHashesMap.set(path, new Set());
+    }
+    if (forceWrite !== '') {
+      this.filePathToHashesMap.get(path)!.add(forceWrite);
+    } else if (node.attribs!.id) {
+      this.filePathToHashesMap.get(path)!.add(node.attribs!.id);
+    }
+  }
+
+  printFilePathToHashesMap() {
+    let result = '';
+    this.filePathToHashesMap.forEach((hashes, filePath) => {
+      result += `${filePath}:\n`;
+      hashes.forEach((hash) => {
+        result += `  ${hash}\n`;
+      });
+    });
+    return result;
+  }
+
+  maintainInclude(node: MbNode, cwf: string) {
```

Comment:
```suggestion
  maintainHashesForInclude(node: MbNode, cwf: string) {
```

Reply:
Thank you for the suggested name. Have commited it, and the suggested name is indeed much clearer